index.bs

```diff
@@ -5313,22 +5313,32 @@ The following types are known as <dfn id="dfn-integer-type" export>integer types
 {{unsigned short}},
 {{long}},
 {{unsigned long}},
-{{long long}} and
-{{unsigned long long}}.
+{{long long}},
+{{unsigned long long}},
+any [=nullable types=] whose [=nullable types/inner type=] is an [=integer type=],
+and any [=annotated types=] whose [=annotated types/inner type=] is an [=integer type=].
 
 The following types are known as <dfn id="dfn-numeric-type" export>numeric types</dfn>:
 the [=integer types=],
 {{float}},
 {{unrestricted float}},
-{{double}} and
-{{unrestricted double}}.
+{{double}},
+{{unrestricted double}},
+any [=nullable types=] whose [=nullable types/inner type=] is a [=numeric type=],
+and any [=annotated types=] whose [=annotated types/inner type=] is a [=numeric type=].
 
 The <dfn id="dfn-primitive-type" export>primitive types</dfn> are
-{{boolean}} and the [=numeric types=].
+{{boolean}},
```

Comment:
Again, this leads to the slightly odd situation that the `PrimitiveType` production no longer matches the set of "primitive types".  Further, the prose around ConstType says:

> The type of a constant (matching ConstType) must not be any type other than a primitive type.

with that last it linking to this definition but it would now actually means a stronger restriction than what that link provides.